2.7/Dockerfile

```diff
@@ -33,16 +33,16 @@ LABEL summary="$SUMMARY" \
       usage="s2i build https://github.com/sclorg/s2i-python-container.git --context-dir=2.7/test/setup-test-app/ centos/python-27-centos7 python-sample-app" \
       maintainer="SoftwareCollections.org <sclorg@redhat.com>"
 
-RUN yum install -y centos-release-scl && \
-    INSTALL_PKGS="libjpeg-turbo libjpeg-turbo-devel python27 python27-python-devel python27-python-setuptools \ 
-	python27-python-pip nss_wrapper httpd24 httpd24-httpd-devel httpd24-mod_ssl httpd24-mod_auth_kerb \
-        httpd24-mod_ldap httpd24-mod_session atlas-devel gcc-gfortran libffi-devel libtool-ltdl enchant" && \
-    yum install -y --setopt=tsflags=nodocs --enablerepo=centosplus $INSTALL_PKGS && \
+RUN INSTALL_PKGS="libjpeg-turbo libjpeg-turbo-devel python27 python27-python-devel python27-python-setuptools \
+        python27-python-pip nss_wrapper httpd24 httpd24-httpd-devel httpd24-mod_ssl \
+        httpd24-mod_auth_kerb httpd24-mod_ldap httpd24-mod_session atlas-devel gcc-gfortran \
+        libffi-devel libtool-ltdl enchant" && \
+    yum install -y centos-release-scl && \
+    yum -y --setopt=tsflags=nodocs install --enablerepo=centosplus $INSTALL_PKGS && \
     rpm -V $INSTALL_PKGS && \
-    # Remove centos-logos (httpd dependency, ~20MB of graphics) to keep image
-    # size smaller.
+    # Remove centos-logos (httpd dependency) to keep image size smaller.
     rpm -e --nodeps centos-logos && \
-    yum clean all -y
+    yum -y clean all --enablerepo='*'
```

Comment:
Why `--enablerepo='*'`?

Reply:
This is how cleancache command is defined in distgen tool https://github.com/devexp-db/distgen/blob/master/distgen/commands.py#L72. I simply call it's method in the template https://github.com/sclorg/s2i-python-container/pull/259/commits/cabe677976b28eaca14f26d952fce561a7715460#diff-bd0e57c1ca2f20c05a1456d4512e2eafR55.

Reply:
I think difference is mainly in rhel7 Dockerfile.

For first `yum` call default repositories are enabled - so metadata are downloaded. Then `prepare-yum-repositories` disable these repositories... And `yum clean` cleans only enabled repositories, that's why `--enablerepo='*'` is needed.